### PR TITLE
Setup authentication on server

### LINKED ISSUE #64: Implement authentication on server.

### PR DESCRIPTION
**Pull request by dting**
Very basic authentication using passport
and passport-facebook.

TODO: We still need to match users that authenticate
with stored emails of placeholder users before creating
a new User instance for them.

Resolves #64

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,26 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
 
 ### Installing Dependencies (tbd)
 
+### Setting Up passport-facebook
+
+1. Sign up to be a [facebook developer](https://developers.facebook.com/).
+2. Click the "Add a New App" on your [apps page](https://developers.facebook.com/apps/).
```
Comment (hannahhenderson) [inline README.md]: The button is called "Create a New App" (the subsequent pop-up is titled "Add a New App")

File: README.md [hunk 2]
```diff
@@ -50,6 +50,26 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+3. Choose a "Website" app and give it a name.
+4. Add a contact email and a category when prompted.
+5. Complete the captcha and wait for the app to be created.
+6. Under the "Tell us about your website" portion of the next page, enter:
```
Comment (hannahhenderson) [inline README.md]: You need to scroll down to the "Tell us about your website" portion, while the "Skip Quick Start" button is visible in the upper right hand corner of prominent dark grey "Quick Start for Website" box.

File: README.md [hunk 3]
```diff
@@ -50,6 +50,26 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+### Setting Up passport-facebook
+
+1. Sign up to be a [facebook developer](https://developers.facebook.com/).
+2. Click the "Add a New App" on your [apps page](https://developers.facebook.com/apps/).
+3. Choose a "Website" app and give it a name.
+4. Add a contact email and a category when prompted.
+5. Complete the captcha and wait for the app to be created.
+6. Under the "Tell us about your website" portion of the next page, enter:  
+     `http://localhost:3000/`
+7. Click the "Skip Quick Start" button.
```
Comment (hannahhenderson) [inline README.md]: Scroll back up and click on the "Skip Quick Start" button

File: README.md [hunk 4]
```diff
@@ -50,6 +50,26 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+8. Click the "+ Add Product" button on the left side of the dashboard.
+9. Click the "Get Started" button next to "Facebook Login".  
+10. For "Valid OAuth redirect URIs", enter:  
+     `http://localhost:3000/auth/facebook/callback`
+11. Click "Save Changes" on the bottom right.
+12. Go back to the app dashboard by clicking the "Dashboard" menu item on the top left.
+13. If you haven't already, make a copy of `./server/config/.env.sample.js` as `./server/config/.env.js`.
```
Comment (hannahhenderson) [inline README.md]: Don't need to say "if you haven't already"

Comment (hannahhenderson) [inline README.md]: Add the "Note, this is a JavaScript file" comment with this line.

File: README.md [hunk 5]
```diff
@@ -50,6 +50,26 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+14. Copy the "App ID" and replace `FACEBOOK_ID`'s value with the copied value. Notice that the `.env.js` file is a javascript file. The value for each key in this file should be a string so make sure there are single quotes around the pasted in value.
```
Comment (hannahhenderson) [inline README.md]: Copy the "App ID" and replace FACEBOOK_ID with the copied value. The value for each key should be a string, so make sure there are single quotes around the pasted in value.

**Comment by hannahhenderson** [comment]
Made some suggestions to the language of the instructional steps, otherwise looks good to go

File: README.md (hunk 5 above)
Comment (hannahhenderson) [inline README.md]: Add a 15th step similar to: While running grunt, navigate to Localhost:3000/auth/facebook and look at the response in your browser. A cookie from the server with a token value should be displayed.

File: README.md (hunk 1 above)
Comment (dting) [inline README.md]: Done.

File: README.md (hunk 2 above)
Comment (dting) [inline README.md]: Done.

File: README.md (hunk 3 above)
Comment (dting) [inline README.md]: Done.

File: README.md (hunk 4 above)
Comment (dting) [inline README.md]: Done.

File: README.md (hunk 5 above)
Comment (dting) [inline README.md]: Done.

**Comment by dting** [comment]
Thanks looking over the changes/instructions. I've made the suggested changes. Let me know if that looks okay to merge. Thanks!

File: README.md [hunk 6]
```diff
@@ -50,6 +50,27 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+6. Scroll down to the "Tell us about your website" portion of the next page, enter:  
+     `http://localhost:3000/`
+7. Click "Next" and scroll back to the top of the page. Click on the "Skip Quick Start" button.
+8. Click the "+ Add Product" button on the left side of the dashboard.
+9. Click the "Get Started" button next to "Facebook Login".  
+10. For "Valid OAuth redirect URIs", enter:  
+     `http://localhost:3000/auth/facebook/callback`
+11. Click "Save Changes" on the bottom right.
+12. Go back to the app dashboard by clicking the "Dashboard" menu item on the top left.
+13. Make a copy of `./server/config/.env.sample.js` as `./server/config/.env.js`. Notice that the `.env.js` file is a javascript file. The value for each property of the exported object in this file should be a string. Make sure there are single quotes around the pasted in values.
```
Comment (hannahhenderson) [inline README.md]: JavaScript should be capitalized

File: README.md [hunk 7]
```diff
@@ -50,6 +50,27 @@ Divvy allows users to enter expenses as they are incurred, and calculates the be
...
+14. Copy the "App ID" and replace `FACEBOOK_ID`'s value with the copied value.
+15. Click the "Show" button for "App Secret". Enter your facebook password to display the value. Copy the shown value and into the `.env.js` file as the `FACEBOOK_SECRET`.
```
Comment (hannahhenderson) [inline README.md]: Facebook should be capitalized

**Pull request merged**